src/archetype.rs

```diff
@@ -26,13 +26,20 @@ use crate::borrow::AtomicBorrow;
 use crate::query::Fetch;
 use crate::{Access, Component, Query};
 
+/// A HashMap with TypeId keys
+///
+/// Because TypeId is already a fully-hashed u64 (including data in the high seven bits,
+/// which hashbrown needs), there is no need to hash it again. Instead, this uses the much
+/// faster no-op hash.
+pub type TypeIdMap<V> = HashMap<TypeId, V, nohash_hasher::BuildNoHashHasher<u64>>;
```

Comment:
I was wondering if you would want to export this type so that someone like @cart could use it. bevy has various `HashMap<TypeId>` that could benefit, although it also `HashSet<TypeId>`, which would be a bit weird to export from hecs since you don't use it.

If you don't want to export this I can explicitly make it `pub(crate)` to be more clear.

Reply:
I'm disinclined to export it since 1. it's trivial and 2. it's out of scope for hecs proper. In principle this is a case for a shared crate, though, it's so trivial that I dunno if it's justified.